src/main/java/hng_java_boilerplate/user/controller/UserController.java

```diff
@@ -30,12 +30,24 @@ public ResponseEntity<?> getUserDetails(@PathVariable String userId) {
     @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
     @GetMapping(value = "/members", produces = "application/json")
     public ResponseEntity<?> getAllMembers(@RequestParam int page, Authentication authentication) {
+        if (authentication == null) return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                    .body(Response.builder().message("Unauthorized").status_code("401").build());
+
         List<MembersResponse> allUsers = userService.getAllUsers(page, authentication);
         Response<?> response = Response.builder().message("Users List Successfully Fetched").status_code("200").data(allUsers).build();
         return ResponseEntity.status(HttpStatus.OK).body(response);
     }
 
 
+    @PreAuthorize("hasRole('ROLE_SUPER_ADMIN')")
+    @DeleteMapping("/delete")
+    public ResponseEntity<?> deleteUserByEmail(@RequestBody DeleteUserRequest request, Authentication authentication) {
+        if (authentication == null) throw new UnAuthorizedException("Unauthorized");
+        Response<?> response = userService.deleteUserByEmail(request, authentication);
+        return ResponseEntity.status(HttpStatus.OK).body(response);
+    }
+
+
     @GetMapping("/me/{id}")
```

Comment:
What is this endpoint supposed to return?

Reply:
Retrieve a single user object

Reply:
if it's just retrieving a single user then the route should be /{id}